### ISSUE TITLE
COPY --from=$ARGUMENT warns with DL3022

### ISSUE THREAD

**Issue by akostadinov** [hadolint/hadolint#933]
- [x] This is a bug report
- [ ] This is a feature request
- [x] I searched existing issues before opening this one

### Expected behavior
No warning

### Actual behavior
```
Containerfile.bug:10 DL3022 warning: COPY --from should reference a previously defined FROM alias
```

### Steps to reproduce the behavior
1. Create a Containerfile as below
2. run `hadolint Containerfile.bug`

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version` or
  `docker run --rm ghcr.io/hadolint/hadolint hadolint --version`:**

```bash
hadolint-2.8.0-8.fc36.x86_64
```

**Dockerfile** (if relevant)

```Dockerfile
ARG PORTA_IMAGE=builder

FROM registry.access.redhat.com/ubi7-minimal as builder
WORKDIR /
RUN echo generate something > foo.txt

FROM registry.access.redhat.com/ubi7-minimal
ARG PORTA_IMAGE
WORKDIR /
COPY --from=$PORTA_IMAGE foo.txt bar.txt
```

**Additional environment details (OS, stack version, etc.)**
Fedora 36

-- related but not same as to #859, #718, #197, #815, #421, #271
